manager/processors/duplicate_tmplvars.processor.php

```diff
@@ -12,49 +12,31 @@
 }
 
 // count duplicates
-$name = $modx->getDatabase()->getValue($modx->getDatabase()->select('name', $modx->getDatabase()->getFullTableName('site_tmplvars'), "id='{$id}'"));
-$count = $modx->getDatabase()->getRecordCount($modx->getDatabase()->select('name', $modx->getDatabase()->getFullTableName('site_tmplvars'), "name LIKE '{$name} {$_lang['duplicated_el_suffix']}%'"));
+$tmplvar = EvolutionCMS\Models\SiteTmplvar::findOrFail($id);
+$name = $tmplvar->name;
+$count = EvolutionCMS\Models\SiteTmplvar::where('name', 'like', $name.' '.$_lang['duplicated_el_suffix'].'%')->count();
 if($count>=1) $count = ' '.($count+1);
 else $count = '';
 
-// duplicate TV
-$newid = $modx->getDatabase()->insert(
-	array(
-		'type'=>'',
-		'name'=>'',
-		'caption'=>'',
-		'description'=>'',
-		'default_text'=>'',
-		'elements'=>'',
-		'rank'=>'',
-		'display'=>'',
-		'display_params'=>'',
-		'category'=>'',
-		), $modx->getDatabase()->getFullTableName('site_tmplvars'), // Insert into
-	"type, CONCAT(name, ' {$_lang['duplicated_el_suffix']}{$count}') AS name, CONCAT(caption, ' Duplicate{$count}') AS caption, description, default_text, elements, rank, display, display_params, category", $modx->getDatabase()->getFullTableName('site_tmplvars'), "id='{$id}'"); // Copy from
 
+$newTmplvar = $tmplvar->replicate();
+$newTmplvar->name = $tmplvar->name.' '.$_lang['duplicated_el_suffix'].$count;
+$newTmplvar->caption = $tmplvar->caption.' Duplicate '.$count.'';
+$newTmplvar->push();
 
-// duplicate TV Template Access Permissions
-$modx->getDatabase()->insert(
-	array(
-		'tmplvarid'=>'',
-		'templateid'=>'',
-		'rank'=>'',
-		), $modx->getDatabase()->getFullTableName('site_tmplvar_templates'), // Insert into
-	"'{$newid}', templateid, rank", $modx->getDatabase()->getFullTableName('site_tmplvar_templates'), "tmplvarid='{$id}'"); // Copy from
-
-// duplicate TV Access Permissions
-$modx->getDatabase()->insert(
-	array(
-		'tmplvarid'=>'',
-		'documentgroup'=>'',
-		), $modx->getDatabase()->getFullTableName('site_tmplvar_access'), // Insert into
-	"'{$newid}', documentgroup", $modx->getDatabase()->getFullTableName('site_tmplvar_access'), "tmplvarid='{$id}'"); // Copy from
+foreach ($tmplvar->tmplvarTemplate as $tmplvarTemplate) {
+    $field = $tmplvarTemplate->toArray();
```

Comment:
the **toArray** method returns data with a relationships. You must use the **attributesToArray** method.